tests/components/border_editor.test.ts

```diff
@@ -0,0 +1,229 @@
+    expect(getBorder(model, "B2")).toBeNull();
+  });
+
+  test("Can set only bottom borders", async () => {
+    setSelection(model, ["A1:B2"]);
+    await setDefaultBorder("bottom");
+
+    expect(getBorder(model, "A2")).toEqual({ bottom: DEFAULT_BORDER_DESC });
+    expect(getBorder(model, "B2")).toEqual({ bottom: DEFAULT_BORDER_DESC });
+    expect(getBorder(model, "A1")).toBeNull();
+    expect(getBorder(model, "B1")).toBeNull();
+  });
+
+  test("Can set only inside hv borders", async () => {
+    setSelection(model, ["A1:B2"]);
+    await setDefaultBorder("hv");
+
+    expect(getBorder(model, "A1")).toEqual({
+      right: DEFAULT_BORDER_DESC,
+      bottom: DEFAULT_BORDER_DESC,
+    });
+    expect(getBorder(model, "A2")).toEqual({
+      right: DEFAULT_BORDER_DESC,
+      top: DEFAULT_BORDER_DESC,
+    });
+    expect(getBorder(model, "B1")).toEqual({
+      left: DEFAULT_BORDER_DESC,
+      bottom: DEFAULT_BORDER_DESC,
+    });
+    expect(getBorder(model, "B2")).toEqual({
+      left: DEFAULT_BORDER_DESC,
+      top: DEFAULT_BORDER_DESC,
+    });
+  });
+
+  test("Can set only inside hv borders", async () => {
+    setSelection(model, ["A1:B2"]);
+    await setDefaultBorder("h");
+
+    expect(getBorder(model, "A1")).toEqual({ bottom: DEFAULT_BORDER_DESC });
+    expect(getBorder(model, "A2")).toEqual({ top: DEFAULT_BORDER_DESC });
+    expect(getBorder(model, "B1")).toEqual({ bottom: DEFAULT_BORDER_DESC });
+    expect(getBorder(model, "B2")).toEqual({ top: DEFAULT_BORDER_DESC });
+  });
+
+  test("Can set only inside hv borders", async () => {
+    setSelection(model, ["A1:B2"]);
+    await setDefaultBorder("v");
+
+    expect(getBorder(model, "A1")).toEqual({ right: DEFAULT_BORDER_DESC });
+    expect(getBorder(model, "A2")).toEqual({ right: DEFAULT_BORDER_DESC });
+    expect(getBorder(model, "B1")).toEqual({ left: DEFAULT_BORDER_DESC });
+    expect(getBorder(model, "B2")).toEqual({ left: DEFAULT_BORDER_DESC });
+  });
+
+  test("Can set only inside hv borders", async () => {
+    setSelection(model, ["A1:B2"]);
+    await setDefaultBorder("external");
+
+    expect(getBorder(model, "A1")).toEqual({
+      left: DEFAULT_BORDER_DESC,
+      top: DEFAULT_BORDER_DESC,
+    });
+    expect(getBorder(model, "A2")).toEqual({
+      left: DEFAULT_BORDER_DESC,
+      bottom: DEFAULT_BORDER_DESC,
+    });
+    expect(getBorder(model, "B1")).toEqual({
+      right: DEFAULT_BORDER_DESC,
+      top: DEFAULT_BORDER_DESC,
+    });
+    expect(getBorder(model, "B2")).toEqual({
+      right: DEFAULT_BORDER_DESC,
+      bottom: DEFAULT_BORDER_DESC,
+    });
+  });
+
+  test("Can set all borders", async () => {
+    setSelection(model, ["A1:B2"]);
+    await setDefaultBorder("clear");
+
+    for (const cell of ["A1", "A2", "B1", "B2"]) {
+      expect(getBorder(model, cell)).toBeNull();
+    }
+  });
+
+  test.each(["thin", "medium", "thick", "dotted", "dashed"])(
+    "Can change borders type",
+    async (style: String) => {
```

Comment:
use the primitive type instead of the object constuctor
```suggestion
    async (style: string) => {
```